### PR TITLE
WIP: Extend debug mode to --show-bears

### PR DESCRIPTION
**Pull request by tulphoon**
Closes https://github.com/coala/coala/issues/4617

### CODE REVIEW

File: tests/TestUtilities.py [hunk 1]
```diff
@@ -66,3 +66,26 @@ class PseudoPlugin:
...
                              return_value=[EntryPoint()]) as mocked:
         yield
+
+
+@contextmanager
+def collect_error_test_bears_module():
+    """
+    This function mocks the ``pkg_resources.iter_entry_points()``
+    to use the testing bear module we have. Hence, it doesn't test
+    the collection of entry points.
```
Comment (userzimmermann) [inline tests/TestUtilities.py]: Please include some reason for this additional `test_bears_module` here

File: tests/coalaTest.py [hunk 2]
```diff
@@ -162,6 +163,16 @@ def test_show_all_bears(self, debug=False):
     def test_show_all_bears_debug(self):
         return self.test_show_all_bears(debug=True)
 
+    def test_show_all_bears_debug_exception(self, debug=True):
```
Comment (userzimmermann) [inline tests/coalaTest.py]: Please use naming like `test_show_all_bears_error` or `test_show_all_bears_error_debug`. It's just more consistent

BTW: Is there already some test for error on collecting bears (without debug stuff)

File: tests/coalaTest.py [hunk 3]
```diff
@@ -162,6 +163,16 @@ def test_show_all_bears(self, debug=False):
...
+        with collect_error_test_bears_module():
+            retval, stdout, stderr = execute_coala(
+                coala.main, 'coala', '-B', '-I', debug=debug)
+            self.assertEqual(retval, 0)
+            # All bears plus 1 line holding the closing colour escape sequence.
+            self.assertEqual(stdout.strip().splitlines(),
+                             TEST_BEARS_COUNT + 1)
+            self.assertFalse(stderr)
```
Comment (userzimmermann) [inline tests/coalaTest.py]: There is no exception at all :)

File: tests/collect_error_test_bears/CollectErrorTestBear.py [hunk 4]
```diff
@@ -0,0 +1,3 @@
+class CollectErrorTestBear():
+    def run(self):
+        pass
```
Comment (userzimmermann) [inline tests/collect_error_test_bears/CollectErrorTestBear.py]: This module must raise some exception :)

**Comment by userzimmermann** [comment]
Taking this over according to https://github.com/coala/coala/issues/4617#issuecomment-340164228

**Comment by shreyans800755** [comment]
@tulphoon @userzimmermann 
If @userzimmermann  is going to work on this, please close this PR if not required as its no longer to be reviewed.

**Comment by aashutoshrathi** [comment]
Resolve conflicts

**Comment by Vedarth** [comment]
This pr should be closed as the issue has been already resolved by another pr  https://github.com/coala/coala/pull/4821

**Comment by Vedarth** [comment]
unack https://github.com/coala/coala/pull/4649/commits/f38fd43e524b22917d255229326096b3ecd8bcb9

**Comment by userzimmermann** [comment]
@Vedarth https://github.com/coala/coala/pull/4821 doesn't solve anything of what this PR is about :) The underlying issues https://github.com/coala/coala/issues/4495 and https://github.com/coala/coala/issues/4617 are unrelated. The only connection here is that this PR would have enabled more comprehensive testing possibilities for https://github.com/coala/coala/pull/4821

@shreyans800755 I will close this as soon as I will have opened a new PR based on @tulphoon's commit :)

**Comment by userzimmermann** [comment]
@Vedarth BTW: You don't need to `unack` what was never `ack`'d ;)

**Comment by userzimmermann** [comment]
@Vedarth But anyway: Thanks for reminding me of this :) And for going through our old stuff and trying to clean up our mess a bit ;) Seriously!

**Comment by nemaniarjun** [comment]
@Makman2  According to comments, this issue has been resolved. This PR can be closed :D

**Comment by nemaniarjun** [comment]
@Makman2  Lets close this PR, as per the comment, https://github.com/coala/coala/issues/4617#issuecomment-340164228

**Comment by jayvdb** [comment]
Closing as author has given up on the patch and re-assigned it to someone else.

**Pull request closed without merging**